src/ssl.c

```diff
@@ -39412,7 +39487,54 @@ void* wolfSSL_SESSION_get_ex_data(const WOLFSSL_SESSION* session, int idx)
 #ifdef HAVE_EX_DATA
     session = ClientSessionToSession(session);
     if (session != NULL) {
-        return wolfSSL_CRYPTO_get_ex_data(&session->ex_data, idx);
+#ifndef NO_SESSION_CACHE
+        if (!session->ownExData) {
```

Comment:
This block of code looks very similar to the one above at 39414-39457. Could this be a new static function?

Reply:
Refactored into `SESSION_ex_data_cache_update`.